libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3286,18 +3286,33 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			(memberName == "min" ||	memberName == "max")
 		)
 			annotation.isPure = true;
-		else if (magicType->kind() == MagicType::Kind::Block && memberName == "chainid" && !m_evmVersion.hasChainID())
+		else if (magicType->kind() == MagicType::Kind::Block)
+		{
+			if (memberName == "chainid" && !m_evmVersion.hasChainID())
 			m_errorReporter.typeError(
 				3081_error,
 				_memberAccess.location(),
 				"\"chainid\" is not supported by the VM version."
 			);
-		else if (magicType->kind() == MagicType::Kind::Block && memberName == "basefee" && !m_evmVersion.hasBaseFee())
-			m_errorReporter.typeError(
-				5921_error,
-				_memberAccess.location(),
-				"\"basefee\" is not supported by the VM version."
-			);
+			else if (memberName == "basefee" && !m_evmVersion.hasBaseFee())
+				m_errorReporter.typeError(
+					5921_error,
+					_memberAccess.location(),
+					"\"basefee\" is not supported by the VM version."
+				);
+			else if (memberName == "difficulty" && m_evmVersion.hasPrevRandao())
+				m_errorReporter.warning(
```

Comment:
I would say this should be an error - because the semantic of `difficulty` vs. `prevrandao` is different. There is no case where `difficulty` in `Paris` means `difficulty` - that's why I would argue that compilation need to fail here.

Reply:
A bit late to [the discussion](https://github.com/ethereum/solidity/issues/13512#issuecomment-1246792479) I'd say :)
I don't really mind either way, but as I understand it, this all already has been discussed

Reply:
The current plan is to make it a warning, since making it an error right away is borderline breaking (even though one can argue about that). With 0.9 it should definitely become an error.

Reply:
@ekpyron yeah that makes sense! @Marenz is that tracked somewhere somehow, so we will not forget it?